dstep/translator/Type.d

```diff
@@ -201,13 +201,18 @@ package string reduceAlias(Type type)
         case "u32": return "uint";
         case "u64": return "ulong";
 
-        default: return null;
+        default: break;
     }
+
+    if (type.spelling == "byte" && translateType(context, type.canonical.kind) == "ubyte")
```

Comment:
Why not just return what `translateType` returns?

Reply:
Because then dstep would incorrectly translated rubbish as `typedef unsigned int byte` to `ubyte`.

Reply:
Sorry, to `uint`.

Reply:
Hmm, right.

Reply:
Technically the above switch statement should check the type kind as well. I'm wondering if we should replace the whole implementation of this function with an associative array, mapping a tuple of the type kind and the spelling to the result.

Reply:
Something like:

```d
enum aliasMapping = [
    tuple("byte", CXTypeKind.uChar): "ubyte",
    .
    .
    .
]

return aliasMapping[tuple(type.spelling, type.canonical.kind)];
```

Reply:
Do you have in minds other reductions than byte -> ubyte. Any other type (short, int, long) cannot be used as this as it's keyword in C and cannot be used as typedef.

Reply:
I thought about this but it looks like byte is the only special case.

Reply:
All the typedefs in the above switch statement. Currently it just checks the spelling without checking the underlying type.

Reply:
I have some problems with typedefs to `long` and `unsigned long` as they can be 32/64 bit... Windows tests are failing : |